### PR TITLE
Closes #1462 Optimize javascript behaviors for full width elements.

### LINKED ISSUE #1462: Optimize javascript behaviors for full width elements.
In. modules/custom/az_paragraphs/js/az_paragraphs_full_width.es6.js

Convert all objects to methods, create new method with attach.
Call new methods in the window event.

Replace usage of IDs with Classes

### PR DESCRIPTION
**Pull request by trackleft**
## Description
Refactor javascript for full-width elements.
Add additional event listener for youtube iframe ready.

## Related issues
#1462 

## How to test
Any page with both a full-width element link a text on media or text on background paragraph, AND one or two sidebars should be affected by this PR.

In any browser you deem supportable, add a page with a full-width paragraph AND other paragraphs. Then add a block or two to either side of the page.

The sidebar should appear beneath the last full-width element on the page.
The rows of all of the full width paragraphs should be the width of the page container.

### Arizona Quickstart (install profile, custom modules, custom theme)
- Patch release changes
   - [ ] Bug fix
   - [x] Accessibility, performance, or security improvement
   - [ ] Critical institutional link or brand change
- Minor release changes
   - [ ] New feature
   - [ ] Breaking or visual change to existing behavior
   - [ ] Non-critical brand change
   - [ ] New internal API or API improvement with backwards compatibility
   - [ ] Risky or disruptive cleanup to comply with coding standards
   - [ ] High-risk or disruptive change (requires upgrade path, risks regression, etc.)
- [x] Other or unknown

### Drupal core
- Patch release changes
   - [ ] Security update
   - [ ] Patch level release (non-security bug-fix release)
   - [ ] Patch removal that's no longer necessary
- Minor release changes
   - [ ] Major or minor level update
- [ ] Other or unknown

### Drupal contrib projects
- Patch release changes
   - [ ] Security update
   - [ ] Patch or minor level update
   - [ ] Add new module
   - [ ] Patch removal that's no longer necessary
- Minor release changes
   - [ ] Major level update
- [ ] Other or unknown

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: modules/custom/az_paragraphs/js/az_paragraphs_full_width.es6.js [hunk 1]
```diff
@@ -1,62 +1,97 @@
+/**
+ * @file
+ * Provides helper functions to ensure proper display of full-width-paragraphs.
+ *
```
Comment (joeparsons) [inline modules/custom/az_paragraphs/js/az_paragraphs_full_width.es6.js:4]:
```suggestion

```

**Review by joeparsons — APPROVED** [review]
These changes look good to me.  I like the way the JS is organized a lot better.

**Review by bberndt-uaz — DISMISSED** [review]
Tested with the [Text on Media](https://7b5e2985-cedf-4a13-ab91-3a35916c835d--pr-1467.probo.build/pages/text-media) page: in Chrome and Safari, the sidebar is reliably moved to the proper place at the bottom of the page. In Firefox, the sidebar is still placed further up the page than it should be. I still think the Text on Media page is special case and this PR seems like a good improvement.

**Comment by trackleft** [comment]
Needs work, full width media video backgrounds load after the js is run.

File: modules/custom/az_paragraphs/js/az_paragraphs_full_width.es6.js [hunk 2]
```diff
@@ -1,62 +1,106 @@
...
-    Drupal.behaviors.azParagraphsPushSidebarDown.attach();
+    calculateScrollbarWidth();
+    calculateFullWidthNegativeMargins();
+    pushSidebarsDown();
+  });
+
+  /**
+   * Recalculates values for CSS variables on youtube iframe ready.
+   */
+   window.addEventListener("onYouTubeIframeAPIReady", () => {
```
Comment (joeparsons) [inline modules/custom/az_paragraphs/js/az_paragraphs_full_width.es6.js:100]:
```suggestion
  window.addEventListener("onYouTubeIframeAPIReady", () => {
```

File: modules/custom/az_paragraphs/js/az_paragraphs_full_width.es6.js [hunk 3]
```diff
@@ -1,62 +1,106 @@
...
+    calculateScrollbarWidth();
+    calculateFullWidthNegativeMargins();
+    pushSidebarsDown();
   });
+
```
Comment (joeparsons) [inline modules/custom/az_paragraphs/js/az_paragraphs_full_width.es6.js:105]:
```suggestion
  });

```

File: modules/custom/az_paragraphs/js/az_paragraphs_full_width.es6.js [hunk 4]
```diff
@@ -1,62 +1,106 @@
...
+  /**
+   * Recalculates values for CSS variables on youtube iframe ready.
+   */
+  window.addEventListener("onYouTubeIframeAPIReady", () => {
+    calculateScrollbarWidth();
+    calculateFullWidthNegativeMargins();
+    pushSidebarsDown();
   });
+
```
Comment (trackleft) [inline modules/custom/az_paragraphs/js/az_paragraphs_full_width.es6.js:105]:
```suggestion
```

**Comment by trackleft** [comment]
This is finally ready, thanks for the commit suggestions @joeparsons

**Comment by joeparsons** [comment]
@trackleft would you rather merge this one or your ESlint/prettier config updates PR first?

The JS here will need to get updated to pass the updated coding standards, right?

**Comment by trackleft** [comment]
I think the eslint first, thanks

**Review by joeparsons — DISMISSED** [review]

**Review by smayden — DISMISSED** [review]

File: modules/custom/az_paragraphs/js/az_paragraphs_full_width.js [hunk 5]
```diff
@@ -6,45 +6,50 @@
...
-  });
   window.addEventListener("resize", function () {
-    Drupal.behaviors.calculateFullWidthNegativeMargins.attach();
-    Drupal.behaviors.calculateScrollbarWidth.attach();
-    Drupal.behaviors.azParagraphsPushSidebarDown.attach();
+    calculateScrollbarWidth();
+    calculateFullWidthNegativeMargins();
+    pushSidebarsDown();
+  });
+  window.addEventListener("onYouTubeIframeAPIReady", function () {
```
Comment (trackleft) [inline modules/custom/az_paragraphs/js/az_paragraphs_full_width.js:50]:
```suggestion
  window.addEventListener("onPlayerReady", function () {
```

File: modules/custom/az_paragraphs/js/az_paragraphs_full_width.js [hunk 6]
```diff
@@ -6,45 +6,50 @@
...
-  });
   window.addEventListener("resize", function () {
-    Drupal.behaviors.calculateFullWidthNegativeMargins.attach();
-    Drupal.behaviors.calculateScrollbarWidth.attach();
-    Drupal.behaviors.azParagraphsPushSidebarDown.attach();
+    calculateScrollbarWidth();
+    calculateFullWidthNegativeMargins();
+    pushSidebarsDown();
+  });
+  window.addEventListener("onPlayerReady", function () {
```
Comment (trackleft) [inline modules/custom/az_paragraphs/js/az_paragraphs_full_width.js:50]: if( const videoPlayer =
              thisContainer.getElementsByClassName("az-video-player");) {
videoPlayer.addEventListener('onPlayerReady'), function () {
}    calculateScrollbarWidth();
    calculateFullWidthNegativeMargins();
    pushSidebarsDown();
}

**Comment by joeparsons** [comment]
This has merge conflicts now that we merged the eslint/prettier PR.

**Comment by bberndt-uaz** [comment]
@trackleft I just tested the [Text on Media page](https://ab47602c-06dd-459a-b22c-85505bc5d53d.probo.build/pages/text-media) again: the sidebar was moved to the correct place every time in Chrome, Safari, and Firefox. Well done!

**Review by bberndt-uaz — DISMISSED** [review]

**Review by danahertzberg — APPROVED** [review]
Looks nice on my basic testing! A great follow up ticket would be to add a delay/fade in of the region content to avoid it flashing up at the top before it's moved.

**Review by bberndt-uaz — APPROVED** [review]

**Review by joeparsons — APPROVED** [review]

**Review by tadean — APPROVED** [review]

**Pull request merged**